File: Java/trees/valid_bst.java
public class valid_bst {

    public boolean isValidBST(TreeNode root){
        return isValid(root,Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY);
    }
    public static boolean isValid(TreeNode node, Double left, Double right){
        if(node == null){
            return true;
        }
        if(!(node.val > left && node.val < right)){
            return false;
        }
        return isValid(node.left,left, (double)node.val) &&
                isValid(node.right, (double)node.val, right);
    }
}

File: Python/trees/valid_bst.py
from typing import Optional

from TreeNode import TreeNode
class Solution:
    def isValidBST(self,root: Optional[TreeNode]) ->bool:

        def isValid(node,left,right):
            if not node:
                return True
            if not(node.val < right and node.val > left):
                return False
            return isValid(node.left,left,node.val) and isValid(node.right,node.val, right)
        return isValid(root, float("-inf"), float("inf"))
